### PR TITLE
HackStudio: Detach from debugged process before terminating

### LINKED ISSUE #3308: HackStudio: Unkillable after debugging sample
Steps to reproduce:
1. Open HackStudio
2. Build the sample app: Build->Build
3. Set break point in the main function
4. Debug the sample app: Build->Debug
5. Continue the sample: Build->Run
6. Sample should have exited successfully
7. Close HackStudio: HackStudio->Quit
8. HackStudio won't terminate and can't be killed through System Monitor, either

Apparently it's stuck in `Queued` state somehow.

Screenshot from 2020-08-25 21-15-30

### PR DESCRIPTION
**Pull request by itamar8910**
Fixes #3308 and some issues with icon loading in the FormEditor.

~~Currently a draft, waiting on #3603 to get merged as it's required for this fix to work.~~

### CODE REVIEW

**Comment by tomuta** [comment]
Hm somehow it feels like #3603 also should have addressed this. Which it didn't...

File: DevTools/HackStudio/Debugger/Debugger.cpp [hunk 1]
```diff
@@ -259,4 +267,12 @@ void Debugger::insert_temporary_breakpoint(FlatPtr address)
     m_state.add_temporary_breakpoint(address);
 }
 
+void Debugger::set_requested_debugger_action(DebuggerAction action)
+{
+    pthread_mutex_lock(Debugger::the().continue_mutex());
+    Debugger::the().m_requested_debugger_action = action;
+    pthread_cond_signal(Debugger::the().continue_cond());
+    pthread_mutex_unlock(Debugger::the().continue_mutex());
```
Comment (awesomekling) [inline DevTools/HackStudio/Debugger/Debugger.cpp:275]: Since we are in `Debugger` here already, no need to keep calling `Debugger::the()` :)

Comment (awesomekling) [reply DevTools/HackStudio/Debugger/Debugger.cpp:275]: Also maybe `set_requested_action` is enough for the name, then we don't have to say "debugger" twice when calling it.

**Comment by itamar8910** [comment]
> Hm somehow it feels like #3603 also should have addressed this. Which it didn't...

Yeah. Probably something to do with exiting the main thread while another thread is `wait`ing on something. I'll see if I can reproduce it in a small example.

**Comment by tomuta** [comment]
Ok I think I know why #3603 isn't fixing it. We need to bring back a change that I removed at some point where we need to de-queue threads in certain situations (after calling `Thread::should_die`). This then would allow the threads to run again to clean up the kernel stack and then should be finalized.

**Comment by tomuta** [comment]
I'm not saying these changes are bad, but SystemManager should always be able to kill HackStudio, whether it has a bug or not.

**Pull request merged**